src/core/lease_manager.ts

```diff
@@ -137,8 +141,11 @@ export class LeaseManager {
         `, attempt: ${chalk.cyan(attempt.toString())}/${chalk.cyan(maxAttempts.toString())}`
 
       await sleep(LEASE_ACQUIRE_RETRY_TIMEOUT)
+      lease = await this.tryAcquireLease(username, leaseName, namespace, task, title)
+    }
 
-      return this.acquireLeaseOrRetry(username, leaseName, namespace, task, title, attempt)
+    if (lease) {
+      await this.k8.deleteNamespacedLease(leaseName, namespace)
```

Comment:
it appears we are deleting the lease that has just been acquired earlier. I believe we need to check if the lease is expired or not before attempting to delete it.

```suggestion
    if (!!lease && this.isLeaseExpired(lease)) {
      await this.k8.deleteNamespacedLease(leaseName, namespace)
    }
```

Reply:
ok, understood it now. If we are here, it must be expired, so we delete it.